src/vision/status_quo/barbara_bridges_sync_and_async.md

```diff
@@ -0,0 +1,332 @@
+
+[How To Vision: Status Quo]: ../how_to_vision/status_quo.md
+[the raw source from this template]: https://raw.githubusercontent.com/rust-lang/wg-async-foundations/master/src/vision/status_quo/template.md
+[`status_quo`]: https://github.com/rust-lang/wg-async-foundations/tree/master/src/vision/status_quo
+[`SUMMARY.md`]: https://github.com/rust-lang/wg-async-foundations/blob/master/src/SUMMARY.md
+[open issues]: https://github.com/rust-lang/wg-async-foundations/issues?q=is%3Aopen+is%3Aissue+label%3Astatus-quo-story-ideas
+[open an issue of your own]: https://github.com/rust-lang/wg-async-foundations/issues/new?assignees=&labels=good+first+issue%2C+help+wanted%2C+status-quo-story-ideas&template=-status-quo--story-issue.md&title=
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+### Introducing `block_on`
+
+Barbara is working on the code for [perf.rust-lang.org] and she wants to do a web request to load various intermediate results. She has heard that the `reqwest` crate is quite nice, so she decides to give it a try. She writes up an async function that does her web request:
+
+[perf.rust-lang.org]: https://perf.rust-lang.org/
+
+```rust
+async fn do_web_request(url: &Url) -> Data {
+    ...
+}
+```
+
+She needs to apply this async function to a number of urls. She wants to use the iterator map function, like so:
+
+```rust
+async fn do_web_request(url: &Url) -> Data {...}
+
+fn aggregate(urls: &[Url]) -> Vec<Data> {
+    urls
+        .iter()
+        .map(|url| do_web_request(url))
+        .collect()
+}
+
+fn main() {
+    /* do stuff */
+    let data = aggregate();
+    /* do more stuff */
+}
+```
+
+Of course, since `do_web_request` is an async fn, she gets a type error from the compiler:
+
+```
+error[E0277]: a value of type `Vec<Data>` cannot be built from an iterator over elements of type `impl Future`
+  --> src/main.rs:11:14
+   |
+11 |             .collect();
+   |              ^^^^^^^ value of type `Vec<Data>` cannot be built from `std::iter::Iterator<Item=impl Future>`
+   |
+   = help: the trait `FromIterator<impl Future>` is not implemented for `Vec<Data>`
+```
+
+"Of course," she thinks, "I can't call an async function from a closure." She decides that since she is not overly concerned about performance, so she decides she'll just use a call to [`block_on` from the `futures` crate](https://docs.rs/futures/0.3.14/futures/executor/fn.block_on.html) and execute the function synchronously:
```

Comment:
```suggestion
"Of course," she thinks, "I can't call an async function from a closure." 

### Introducing `block_on`

She decides that since she is not overly concerned about performance, so she decides she'll just use a call to [`block_on` from the `futures` crate](https://docs.rs/futures/0.3.14/futures/executor/fn.block_on.html) and execute the function synchronously:
```